core/tests/test_views.py

```diff
@@ -0,0 +1,659 @@
+from datetime import timedelta
+
+import pytest
+from django.urls import reverse
+from django.utils import timezone
+
+from core.models import (
+    Actor,
+    CinemaHall,
+    Genre,
+    Movie,
+    MovieScreening,
+    Seat,
+    Ticket,
+)
+from core.serializers import (
+    ActorSerializer,
+    CinemaHallSerializer,
+    GenreSerializer,
+    SeatSerializer,
+)
+from core.tests.factories import (
+    ActorFactory,
+    CinemaHallFactory,
+    GenreFactory,
+    MovieFactory,
+    MovieScreeningFactory,
+    SeatFactory,
+    TicketFactory,
+)
+
+
+@pytest.mark.django_db
+class TestGenre:
+    url_list = reverse("genre-list")
+
+    def test_genre_post(self, client):
+        genre = GenreFactory.build()
+        serialized = GenreSerializer(genre).data
+
+        response = client.post(
+            self.url_list, data=serialized, content_type="application/json"
+        )
+        response_data = response.json()
+
+        assert response.status_code == 201
+        assert response_data["name"] == serialized["name"]
+        assert response_data["is_for_adults"] == serialized["is_for_adults"]
+        assert Genre.objects.filter(name=serialized["name"]).exists()
+        assert set(response_data.keys()) == {"id", "name", "is_for_adults"}
+
+    def test_genre_post_blank(self, client):
+        response = client.post(self.url_list, data={}, content_type="application/json")
+
+        assert response.status_code == 400
+        assert response.json() == {"name": ["This field is required."]}
+
+    def test_genre_post_invalid_name(self, client):
+        response = client.post(
+            self.url_list, data={"name": "a" * 51}, content_type="application/json"
+        )
+
+        assert response.status_code == 400
+        assert response.json() == {
+            "name": ["Ensure this field has no more than 50 characters."]
+        }
+
+    def test_genre_get_list(self, client):
+        genre1 = GenreFactory.create()
+        genre2 = GenreFactory.create()
+
+        response = client.get(self.url_list)
+        response_data = response.json()
+
+        assert response.status_code == 200
+        assert len(response_data) == 2
+        assert {"id", "name", "is_for_adults"} <= set(response_data[0].keys())
```

Comment:
same as above, if we have one full serialization test, we can skip that kind of checks. Also this helps for easy test maintenance and single responsibility